### DISCUSSION TITLE
特性请求    添加F2FS文件系统容量维护工具

### DISCUSSION THREAD

**Discussion by virusaur-dino** [immortalwrt/immortalwrt#1300] [Q&A]
### Describe the bug

参考原生版本文档对rootfs进行扩容发现缺乏resize.f2fs工具支持
https://openwrt.org/docs/guide-user/installation/openwrt_x86#expanding_root_filesystem

### ImmortalWrt version

r27625-416c8c5c91

### ImmortalWrt release

23.05.2

### ImmortalWrt target/subtarget

x86/64

### Device

QEMU Standard PC (i440FX + PIIX, 1996)

### Image kind

Official downloaded image

### Steps to reproduce

opkg install losetup resize2fs
 
# Map loop device to root partition
losetup /dev/loop0 /dev/sda2 2> /dev/null
 
# Expand root filesystem
resize2fs -f /dev/loop0

# Output
root@ImmortalWrt:~# resize2fs /dev/loop0
resize2fs 1.47.0 (5-Feb-2023)
resize2fs: Bad magic number in super-block while trying to open /dev/loop0
Couldn't find valid filesystem superblock.

### Terms

- [X] I am reporting an issue for ImmortalWrt, not an unsupported fork.

**Comment by fenghuilee** [comment]
遇到同样的问题

**Comment by cnzgray** [comment]
也有同样的问题，固件是x86的squashfs

**Comment by 1715173329** [comment]
要扩容，最好用 ext4 文件系统，或者利用 imagebuiler 自行调节 rootfs 大小生成映像。
